weave/ops_primitives/boolean.py

```diff
@@ -41,3 +41,11 @@ def none_coalesce(lhs: typing.Any, rhs: typing.Any):
     if isinstance(lhs, list) and isinstance(rhs, list):
         return [l or r for l, r in zip(lhs, rhs)]
     return lhs or rhs
+
+
+@op(output_type=lambda input_type: types.optional(input_type["results"].object_type))
+def cond(cases: dict[str, bool], results: list[typing.Any]):
```

Comment:
why have `cases` be a dict if the keys are ignored?

Reply:
In the final implementation, I changed results to a TypedDict as well. This is because of vectorization. We want each case boolean as a column, and each result as a column. If we use lists these become ListArray under vectorization and we have to reseparate the columns.